edgeless_api/proto/messages.proto

```diff
@@ -53,6 +53,16 @@ message SpawnFunctionRequest {
     StateSpecification state_specification = 6;
 }
 
+// Message to request the creation a new function instance.
+message SpawnFunctionResponse {
+    // If present it means that the request has been rejected.
+    // In this case, other fields may not be present or contain meaningful data.
+    optional ResponseError response_error = 1;
+
+    // The identifier of the newly-spawned function instance, if accepted.
+    optional InstanceId instance_id = 2;
```

Comment:
Do you mean a `oneof`?

I am not particularly fond of that in protobuf because the C++ implementation of that is really stupid and dangerous.

Never used in Rust, we can try. Anyway, it will not guarantee that on-wire there is always at most, just that _if there are two_, the last one will be considered (which IMO is far less than ideal).

Reply:
Yes. Fundamentally it should be clear that there should only be one of them. Since we use Rust as the main language (and in fact the Rust API is the reference while gRPC is just one binding) I would still model that using variants (although I'm not familiar with the C++ behavior).

Reply:
OK, will do that as part of the API update following the 1st hackathon

(just FYI: in C++ the binding does not offer any special guarantee, it just leads to undefined behaviour if you do something like set(A), set(B), use(A) with A,B being in a oneof relation).